### PR TITLE
KOGITO-2348: Scenario fails if DMN model imports another one

### PR DESCRIPTION
**Pull request by yesamer**
**JIRA**: https://issues.redhat.com/browse/KOGITO-2348

Reported case fails with this execption: `Could not locate required dependency while importing DMN for namespace ...`
The cause is: Duplicated DMN files are imported in the working memory, this because original DMN files are copied inside `/target/` folder. This leads to incorrectly evaluate `if` statement in `ImportDMNResolverUtil:44`.
To resolve the issue, I filtered out any DMN and PMML files inside `/target/` folder.
Please not, provided reproducer is not complete (`Should the driver be suspended?` decision doesn't consider `has strict rules` imported data), then I provided a working one in the ticket as attachment.

Changes provided in drools-wb PR resolve another use case (importing of DMNType inside an imported DMN Model).

**Referenced Pull Requests**:
* https://github.com/kiegroup/drools-wb/pull/1428
* https://github.com/kiegroup/kogito-runtimes/pull/789

Case "Canada" is successful: Canada is part of the counties defined in imported `GeneralRules.dmn`
Screenshot from 2020-10-02 12-35-11

Case "Italy" fails: Italy is NOT part of the counties defined in imported `GeneralRules.dmn`
Screenshot from 2020-10-02 12-36-26

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by yesamer** [comment]
jenkins retest this please

**Review by danielezonca — COMMENTED** [review]
Can you please add two DMN in test resources one that imports the other + a scesim file to execute them?

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
@danielezonca Test case provided.

**Review by danielezonca — APPROVED** [review]

File: drools/kogito-scenario-simulation/src/main/java/org/kogito/scenariosimulation/runner/KogitoDMNScenarioRunnerHelper.java [hunk 1]
```diff
@@ -133,4 +133,8 @@ private DMNRuntime initDmnRuntime() {
         }
     }
 
+    private boolean filterResource(Path path, String extension) {
+        return Files.isRegularFile(path) && path.toString().endsWith(extension) && !path.toString().contains("/target/");
```
Comment (jstastny-cz) [inline drools/kogito-scenario-simulation/src/main/java/org/kogito/scenariosimulation/runner/KogitoDMNScenarioRunnerHelper.java:137]: Wouldn't moving the String#contains and String#endsWith to the beginning of the return statement optimize the evaluation of &&? e.g. Based on https://rules.sonarsource.com/java/tag/performance/RSPEC-3725 the Files methods have poor performance, so in theory it's a waste to invoke it on target-located files when those are filtered out in the end anyway.
I would actually reverse the order as is now, filter out any path containing target, get just paths with given extension and only the check the file.

Comment (yesamer) [reply drools/kogito-scenario-simulation/src/main/java/org/kogito/scenariosimulation/runner/KogitoDMNScenarioRunnerHelper.java]: @jstastny-cz Excellent suggestion, done!

**Review by jstastny-cz — APPROVED** [review]
LGTM, thanks

**Comment by yesamer** [comment]
jenkins retest this please

**Review by jstastny-cz — APPROVED** [review]
LGTM

**Pull request merged**